File: Serene/Serene.Script/Imports/FormContexts/Northwind.ProductForm.cs

namespace Serene.Northwind
{
    using Serenity;
    using Serenity.ComponentModel;
    using System;
    using System.Runtime.CompilerServices;

    public partial class ProductForm : PrefixedContext
    {
        [InlineConstant] public const string FormKey = "Northwind.Product";
    
        public ProductForm(string idPrefix) : base(idPrefix) {}
    
        public StringEditor ProductName { [InlineCode("{this}.w('ProductName', Serenity.StringEditor)")] get; private set; }
        public ImageUploadEditor ProductImage { [InlineCode("{this}.w('ProductImage', Serenity.ImageUploadEditor)")] get; private set; }
        public BooleanEditor Discontinued { [InlineCode("{this}.w('Discontinued', Serenity.BooleanEditor)")] get; private set; }
        public LookupEditor SupplierID { [InlineCode("{this}.w('SupplierID', Serenity.LookupEditor)")] get; private set; }
        public LookupEditor CategoryID { [InlineCode("{this}.w('CategoryID', Serenity.LookupEditor)")] get; private set; }
        public StringEditor QuantityPerUnit { [InlineCode("{this}.w('QuantityPerUnit', Serenity.StringEditor)")] get; private set; }
        public DecimalEditor UnitPrice { [InlineCode("{this}.w('UnitPrice', Serenity.DecimalEditor)")] get; private set; }
        public IntegerEditor UnitsInStock { [InlineCode("{this}.w('UnitsInStock', Serenity.IntegerEditor)")] get; private set; }
        public IntegerEditor UnitsOnOrder { [InlineCode("{this}.w('UnitsOnOrder', Serenity.IntegerEditor)")] get; private set; }
        public IntegerEditor ReorderLevel { [InlineCode("{this}.w('ReorderLevel', Serenity.IntegerEditor)")] get; private set; }
    }
}

File: Serene/Serene.Core/Imports/ServerTypings/Northwind.ProductForm.ts
namespace Serene.Northwind {
    export interface ProductForm {
        ProductName: Serenity.StringEditor;
        ProductImage: Serenity.ImageUploadEditor;
        Discontinued: Serenity.BooleanEditor;
        SupplierID: Serenity.LookupEditor;
        CategoryID: Serenity.LookupEditor;
        QuantityPerUnit: Serenity.StringEditor;
        UnitPrice: Serenity.DecimalEditor;
        UnitsInStock: Serenity.IntegerEditor;
        UnitsOnOrder: Serenity.IntegerEditor;
        ReorderLevel: Serenity.IntegerEditor;
    }

    export class ProductForm extends Serenity.PrefixedContext {
        static formKey = 'Northwind.Product';
        private static init: boolean;

        constructor(prefix: string) {
            super(prefix);

            if (!ProductForm.init)  {
                ProductForm.init = true;

                var s = Serenity;
                var w0 = s.StringEditor;
                var w1 = s.ImageUploadEditor;
                var w2 = s.BooleanEditor;
                var w3 = s.LookupEditor;
                var w4 = s.DecimalEditor;
                var w5 = s.IntegerEditor;

                Q.initFormType(ProductForm, [
                    'ProductName', w0,
                    'ProductImage', w1,
                    'Discontinued', w2,
                    'SupplierID', w3,
                    'CategoryID', w3,
                    'QuantityPerUnit', w0,
                    'UnitPrice', w4,
                    'UnitsInStock', w5,
                    'UnitsOnOrder', w5,
                    'ReorderLevel', w5
                ]);
            }
        }
    }
}
